composables/useAuth.ts

```diff
@@ -0,0 +1,62 @@
+import { ref } from 'vue'
+import { storeToRefs } from 'pinia'
+import { useWalletStore } from '~/stores/wallet'
+import { useBookStoreApiStore } from '~/stores/book-store-api'
+
+export function useAuth () {
+  const bookStoreApiStore = useBookStoreApiStore()
+  const store = useWalletStore()
+  const { wallet, signer } = storeToRefs(store)
+  const { connect, disconnect, signMessageMemo } = store
+  const { authenticate, clearSession } = bookStoreApiStore
+  const toast = useToast()
+
+  const isLoading = ref(false)
+
+  async function onClickAuth () {
```

Comment:
great job! but I suggest changing the ref & function name too to prevent duplicated naming in other places
```suggestion
  const isAuthenticating = ref(false)

  async function authenticate () {
```

Reply:
Since `authenticate` is a duplicate name：
[🎨 Rename props](https://github.com/likecoin/nft-book-press/pull/245/commits/4ebb7b6134d6f949aabf5bf0bdb323c7ced9ef22)